discovery_engine/test/integration/user_reacted_test.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+@Timeout(Duration(seconds: 60))
+
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show
+        ClientEventSucceeded,
+        DiscoveryEngine,
+        DocumentId,
+        DocumentsUpdated,
+        EngineEvent,
+        EngineExceptionRaised,
+        EngineExceptionReason,
+        NextFeedBatchRequestSucceeded,
+        UserReaction;
+import '../logging.dart' show setupLogging;
+import 'utils/create_config.dart'
+    show TestEngineData, createConfig, setupTestEngineData;
+import 'utils/db.dart' show loadEngineState;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  group('DiscoveryEngine changeUserReaction', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+
+    setUp(() async {
+      data = await setupTestEngineData();
+    });
+
+    tearDown(() async {
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('change the user reaction of a document', () async {
+      server = await LocalNewsApiServer.start();
+      var engine = await DiscoveryEngine.init(
+        configuration: createConfig(data, server.port),
+      );
```

Comment:
this pattern repeats over several tests, should we encapsulate this in a utility function as well? (but only if this doesn't make too much work, as we don't have a native tuple type to return in dart)

Reply:
Good idea :+1 

I also moved the server + engine (when possible) into the setup function and call `engine.dispose()` at the end.
Please have another look